### ISSUE TITLE
Environment variable setting overrides cannot set values to Boolean false

### ISSUE THREAD

**Issue by tdonohue** [samvera-labs/hyku#1462]
All settings in `config/settings.yml` can be overridden using environment variables, as defined in `config/initializers/config.rb`: https://github.com/samvera-labs/hyku/blob/master/config/initializers/config.rb

However, currently any Settings that default to boolean `true` (e.g. [`admin_only_tenant_creation`](https://github.com/samvera-labs/hyku/blob/master/config/settings.yml#L10)) **cannot be overridden via an environment variable.**  So, in the case of `admin_only_tenant_creation`, creating an environment variable `SETTINGS__MULTITENANCY__ADMIN_ONLY_TENANT_CREATION=false` would have no effect.

This seems to be the result of a bug in the `config` gem which is described here: https://github.com/railsconfig/config/issues/178

Simply put, any environment variable overrides that are set to `false` are ready as "false" (the String), and all Strings are truthy.  So, the String "false" becomes boolean `true`.

After some discussion with @mjgiarlo, we've decided to try to monkey-patch this method : https://github.com/railsconfig/config/blob/3131b54a8d5efbba88dbced80019a5451f04c708/lib/config/options.rb#L191  We'll also send a patch back to the `config` gem in the hopes that it can properly support Booleans in the future.
